File: abc/abc170/abc170_c.cs
// https://atcoder.jp/contests/abc170/submissions/14363134

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var input1 = Console.ReadLine().Split().Select(int.Parse).ToArray();
        var X = input1[0];
        if (input1[1] == 0)
        {
            Console.ReadLine();
            Console.WriteLine(X);
            return;
        }

        var input2 = Console.ReadLine().Split().Select(int.Parse).ToArray();

        var a = new bool[102];
        for (int i = 0; i < input2.Length; i++)
        {
            a[input2[i]] = true;
        }

        var ridx = X;
        while (a[ridx])
        {
            ridx++;
        }
        var lidx = X;
        while (a[lidx])
        {
            lidx--;
        }

        Console.WriteLine((X - lidx) > (ridx - X) ? ridx : lidx);
    }
}

File: abc/abc170/abc170_c.fs
// https://atcoder.jp/contests/abc170/submissions/14389431

let [| X; N |] = stdin.ReadLine().Split() |> Array.map int

let solver =
    let p = stdin.ReadLine()
    if N = 0 then X
    else
        let pi = p.Split() |> Array.map int
        let np = [ 0..102 ] |> List.filter (fun i -> Array.contains i pi |> not)

        let idx =
            np
            |> List.mapi (fun i y -> i, abs (y - X))
            |> List.minBy snd
            |> fst
        np.[idx]

solver |> stdout.WriteLine
